README.md

```diff
@@ -204,13 +204,32 @@ modules.
 sequencer.run();
 ```
 
+Sequencer can be run with a custom progress object
```

Comment:
Just above here, where we introduce `.run()`, let's say:

```
## sequencer.run()

The `.run()` method starts the sequencer, running through each step to process the input image into a final output image. 

There are several different ways to use the `run()` function. You can pass in:

* `index` -- an `integer` - 
* `progressObj` -- an `object` used for a progress-tracking object that will be passed along the sequence (see Progress Tracking, below)
* `callback` -- a `function` to be run upon completion of the sequence
```

They can be used in different combinations:

`sequencer.run(index)` - run the sequencer from the given `index` starting step position

`sequencer.run(progressObj)` - 

...and so on, showing each usage separately (so not mixing `index/progressObj` in one example)

For examples of how `.run()` can be used, see [this test file demonstrating each use](...).
```

What do you think of this? Just really thorough, lots of examples, and link to the tests where we demonstrate it all working? The tests could be moved into a `run.js` test file.

Reply:
@jywarren this is awesome, I can do this, this is really explanatory!! Let's first complete the urlSteps PR then rebase to avoid merge conflicts later since this is on an old fork!!